File: src/Java/501-600/551.StudentAttendanceRecordI.java


import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public boolean checkRecord(String s) {
        int Acount = 0;
        int Lcount = 0;
        for (char c : s.toCharArray()){
            Acount += c == 'A' ? 1 : 0;
            if (Acount >= 2)
                return false;
            Lcount = c == 'L' ? Lcount + 1 : 0; 
            if (Lcount >= 3)
                return false;
        }           
        return true;
    }
}

File: src/Python/501-600/551.StudentAttendanceRecordI.py

class Solution:
    def checkRecord(self, s):
        """
        :type s: str
        :rtype: bool
        """
        Acount = 0
        Lcount = 0
        for c in s:
            Acount += 1 if c == 'A' else 0
            if Acount >= 2:
                return False
            Lcount = Lcount + 1 if c == 'L' else 0 
            if Lcount >= 3:
                return False
        return True
            
if __name__ == '__main__':
    solution = Solution()
    print(solution.checkRecord("PPALLP"))
    print(solution.checkRecord("PPALLL"))
    print(solution.checkRecord("LLLALL"))
else:
    pass
